### PR TITLE
REL-4041: DARP support part I

### PR DESCRIPTION
**Pull request by cquiroz**
This contains DARP support on the p1 model and the PIT. It still needs some work on p1-monitor, skeletons and the backend servers

### CODE REVIEW

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/proposal/ProposalView.scala [hunk 1]
```diff
@@ -114,54 +129,55 @@ class ProposalView(advisor:ShellAdvisor) extends BorderPanel with BoundView[Prop
     add(remove, BorderPanel.Position.East)
 
     // Panel for the buttons
-    object selectPanel extends BorderPanel {panel =>
+    lazy val selectPanel = new BorderPanel {panel =>
```
Comment (swalker2m) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/proposal/ProposalView.scala:132]: were there initialization issues or something?

**Review by swalker2m — APPROVED** [review]
I don't know what DARP is, and I think if I _did_ know I'd be morally opposed, but the code looks good to me.  :+1:   There will be no downstream impact of any of this correct?  Right? Yes?

**Comment by cquiroz** [comment]
There certainly will be downstream impacts in ITAC and the ODB

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/proposal/ProposalView.scala (hunk 1 above)
Comment (cquiroz) [reply bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/proposal/ProposalView.scala:132]: Yes, they are called on the setup of the parent ui control, when they become a `val` (as we have more than one `selectPanel`) the initialization fails unless we declare it `lazy`

**Pull request merged**